package/src/completionCacheManager/completionCacheManager.ts

```diff
@@ -0,0 +1,23 @@
+import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';
+import * as ls from 'vscode-languageserver-types';
+
+export const createCompletionCacheManager = (getFromLanguageService: GetFromLanguageService) => {
+    let completionList: ls.CompletionList | undefined;
+    let lastWord: string | undefined;
+
+    return {
+        getCompletionItems: async (word: string | undefined, resource: monaco.Uri, position: ls.Position) => {
+            const isIncluded = word?.includes(lastWord);
+            const shouldGetItems = !lastWord || !word || !isIncluded;
+            if (shouldGetItems) {
+                completionList = await getFromLanguageService(resource, position);
```

Comment:
If we were on top of things I'd expect to see something after the wait point to check if we're still active or not.

I know our language server doesn't currently have much in the way of cancelation support, so I'm not expecting this PR to fix it.

Reply:
Not sure I understand

Reply:
Actually, I am sure I don't :)

Reply:
For example, if we leave the page completion items are being fetched, we should stop progress.

Right now it's pretty easy to create exceptions by quickly closing the page after doing something to query text when the language server is active.